README.md

```diff
@@ -34,35 +34,62 @@ You can either [clone the repository from GitHub](https://github.com/jtebert/kil
 
 ### Dependencies
 
-[HDF5](https://portal.hdfgroup.org/display/HDF5/HDF5), [SFML](https://www.sfml-dev.org/index.php), and [OpenMP](https://www.openmp.org/). Install on Ubuntu-like systems with: `sudo apt install libsfml-dev libhdf5-dev libomp-dev`
+[HDF5](https://portal.hdfgroup.org/display/HDF5/HDF5), [SFML](https://www.sfml-dev.org/index.php), and [OpenMP](https://www.openmp.org/). Install on Ubuntu-like systems with:
 
-[nlohmann/json](https://github.com/nlohmann/json): Add [`json.hpp`](https://github.com/nlohmann/json/releases/download/v3.5.0/json.hpp) to the `include/` folder. **Note:** This links to the most recent release (3.5) that has been verified for compatibility. You can find other [versions and release notes on Github](https://github.com/nlohmann/json/releases).
+    sudo apt install libsfml-dev libhdf5-dev libomp-dev
 
 ### Compile
 
-Create a static library at `bin/libKilosim.a`:
+To compile, copy and paste the following:
 
-    make
-or
+    mkdir build                                #Create a build directory
+    cd build                                   #Move into build directory
+    cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo .. #Configure for your system
+    make                                       #Build the library
+    make install                               #Install the library to the source directory
+    cd ..                                      #Move back to source directory
 
-    make static
+Alternatively, you can run:
 
-Create an executable at `bin/kilosim` using the example `MyKilobot.cpp` and `test.cpp`:
+    make kilosim                        #Build library
+    make examples                       #Build examples (optional)
+    make kilosim_docs                   #Build documentation (optional)
 
-    make exec
+To clean up the build enter the source directory and run:
 
-**NOTE:** This is tested with g++ 7.3 but currently fails (for as-of-yet unclear reasons) with g++ 8.1
+    rm -rf build bin lib
+
+Flags:
+
+ * `-DSANITIZE_ADDRESS=On`: Use address sanitizer to check memory safety
+
+Use flags like so:
+
+    cmake -DSANITIZE_ADDRESS=ON ..
+
+Compatibility:
+
+ * g++ 5   - doesn't work
+ * g++ 7.3 - works
+ * g++ 8.1 - fails due to compiler issue(?)
+ * g++ 8.3 - works 
 
 ### Run
 
-Run the executable generated by `make exec`
+Run the examples generated by `make examples`
```

Comment:
`make examples` is placing the compiled executable in `build/kilosim_example` instead of in `bin/`

I also tried adding back in the Viewer in the example (to make it as complete as possible as a test case) and got the following error:

```
[ 80%] Built target kilosim
[ 90%] Linking CXX executable kilosim_example
/usr/bin/ld: libkilosim.a(Viewer.cpp.o): undefined reference to symbol '_ZN2sf6Window8setTitleERKNS_6StringE'
//usr/lib/x86_64-linux-gnu/libsfml-window.so.2.4: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
CMakeFiles/kilosim_example.dir/build.make:113: recipe for target 'kilosim_example' failed
make[3]: *** [kilosim_example] Error 1
CMakeFiles/Makefile2:136: recipe for target 'CMakeFiles/kilosim_example.dir/all' failed
make[2]: *** [CMakeFiles/kilosim_example.dir/all] Error 2
CMakeFiles/Makefile2:175: recipe for target 'CMakeFiles/examples.dir/rule' failed
make[1]: *** [CMakeFiles/examples.dir/rule] Error 2
Makefile:203: recipe for target 'examples' failed
make: *** [examples] Error 2
```

Reply:
Seems like this is an issue with it missing `sfml-window` somewhere in the compilation. My next commit should fix this by adding it to the `CMakeLists.txt`

Reply:
Looks like it's working after your commit?

Reply:
Did you try running `make install`? (I've updated the `README.md` and cmake files to require that the installation directory be specified.

Reply:
Yeah, the SFML/`make examples` is compiles now (but is still in the build directory).

What do you mean about specifying the install directory?

Reply:
Per the README, try:
```
    mkdir build                                #Create a build directory
    cd build                                   #Move into build directory
    cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX=../ .. #Configure for your system
    make                                       #Build the library
    make install                               #Install the library to that indicated by `CMAKE_INSTALL_PREFIX` above
    cd ..                                      #Move back to source directory
```
The `CMAKE_INSTALL_PREFIX` sets the directory in which the `bin/` and `lib/` directories get made.

Reply:
I'm running it like the readme says, and it's generating a `bin` directory in the repo root folder, but it's still creating the `kilosim_example` executable in the `build` folder.

Reply:
That's expected: everything gets built in `build` and then copied where it needs to be, so there should be a `kilosim_example` in the `bin/` directory as well.

Reply:
OK, I think I get what's happening now. When I run `make examples`, it generates the `kilosim_example` file in the build directory, but it doesn't get moved/copied to `bin` until you run `make install` *after* running `make examples`, which doesn't seem intuitive.

Reply:
The `examples` target is not built by default since that would mean that any code including kilosim as a library would then have to unnecessarily compile the examples.

Reply:
Can `README` be edited somehow to address your confusion?

Reply:
It makes sense that the default `make` doesn't create the examples.

My confusion is what happens when you run `make examples`. That generate the executable but doesn't put it into the bin folder. My intuition would be that running `make examples` would also deal with copying the output into bin, rather than having to run `make install` after `make examples`.

Does that make sense, or is that a mis-understanding of the conventional compilation pipeline?

Reply:
I can see it the way you describe it, but that isn't the way the tool works. You can think of a project of consisting of potentially many separate pieces, all of which can be made independently. But the install operation takes all the pieces you have chosen to make and installs them at all at once.

Reply:
That makes sense -- just me misunderstanding the way it works. I'll make some edits to the readme to try and clarify (if nothing else for my own sake.)